nasa-gibs/worldview
# WV-2669 autoplay animation via permalink

## Description
1) Include timestamp in lower left corner while in distraction-free mode (using Helvetica font family).

2) Provide ability to auto-play animations on app load via permalink. The new URL parameter is "aa". NOTE: for this to work the URL needs to be loaded with the animation parameters. If you just add `aa=true` to a URL without animation parameters you will not see an animation.

If the URL includes "aa=true" autoplay of animations will be enabled. 
If the URL includes "aa=true" autoplay & df=true (distraction free enabled) animations will autoplay with traditional animation controls. 
If the URL includes "kiosk=true" & df=true (distraction free enabled) the animation control window will be simplified to only show the date range of the animation. 

Fixes # WV-2669

## How To Test
- git checkout wv-2669-autoplay-anim
- npm run watch

This parameter could be applied in a few different situations:
- [Animation requested in normal UI](http://localhost:3000/?v=-233.42574010961158,-215.14699075840838,243.98235687716902,177.06399523009395&z=4&ics=true&ici=5&icd=10&as=2023-04-23-T15%3A50%3A00Z&ae=2023-04-24-T01%3A20%3A00Z&l=Reference_Labels_15m(hidden),Reference_Features_15m(hidden),Coastlines_15m(opacity=0.86),GOES-East_ABI_GeoColor,GOES-West_ABI_GeoColor,Himawari_AHI_Band3_Red_Visible_1km&lg=true&al=true&av=10&ab=on&aa=true&t=2023-04-23-T23%3A00%3A00Z&aa=true); This should display the full UI, full animation controls & autoplay the video.
- [Animation requested in Distraction Free Mode](http://localhost:3000/?v=-233.42574010961158,-215.14699075840838,243.98235687716902,177.06399523009395&z=4&ics=true&ici=5&icd=10&as=2023-04-23-T15%3A50%3A00Z&ae=2023-04-24-T01%3A20%3A00Z&l=Reference_Labels_15m(hidden),Reference_Features_15m(hidden),Coastlines_15m(opacity=0.86),GOES-East_ABI_GeoColor,GOES-West_ABI_GeoColor,Himawari_AHI_Band3_Red_Visible_1km&lg=true&al=true&av=10&ab=on&aa=true&t=2023-04-23-T23%3A00%3A00Z&aa=true&df=true); This should display the limited Distraction Free UI with full animation controls & autoplay the video.
- [Animation requested in Distraction Free Mode with kiosk mode enabled](http://localhost:3000/?v=-233.42574010961158,-215.14699075840838,243.98235687716902,177.06399523009395&z=4&ics=true&ici=5&icd=10&as=2023-04-23-T15%3A50%3A00Z&ae=2023-04-24-T01%3A20%3A00Z&l=Reference_Labels_15m(hidden),Reference_Features_15m(hidden),Coastlines_15m(opacity=0.86),GOES-East_ABI_GeoColor,GOES-West_ABI_GeoColor,Himawari_AHI_Band3_Red_Visible_1km&lg=true&al=true&av=10&ab=on&aa=true&t=2023-04-23-T23%3A00%3A00Z&aa=true&df=true&kiosk=true); This should display the limited Distraction Free UI with reduced animation controls (showing only time frame) & autoplay the video.
 
@nasa-gibs/worldview
